magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala

```diff
@@ -238,11 +238,15 @@ object Registry {
           }
           val totalWeighting = ratings.map(_.weighting).reduce(_ + _)
 
-          if (totalWeighting > 0) {
+          val score = if (totalWeighting > 0) {
             ratings.map(rating =>
               (rating.score) * (rating.weighting / totalWeighting)).reduce(_ + _)
           } else 0d
-        case _ => 0d
+
+          // Make sure no quality score is set to zero, otherwise it results in relevance being * by 0 which makes
+          // results come back in a random order
+          if (score > 0) score else 0.01
+        case _ => 1d
```

Comment:
When no qualityAspect, do we want it to be 1 star or 0 star?
If we want it to be 0 star, here probably should be `case _ => 0.01d`?

Reply:
I was thinking we give aspects with no quality the benefit of the doubt - we assume that they're 100% quality until we know that they're not.

Otherwise
a) we end up ranking datasets that might be perfectly good as 0% quality until a minion catches up and
b) if there's no minion (like on internal instances), all the datasets are 0%. Which I guess is fine as long as it doesn't affect search rankings relevant to each other, but it's still a bit weird IMHO.

Reply:
👍